app/js/search_bar.js

```diff
@@ -0,0 +1,56 @@
+
+import React from "react";
+
+import {useModal} from "./modals/modals";
+import AddRoomModal from './modals/add_room_modal';
+import { create_room } from "./zosia_api";
+
+
+const SearchBar = (props) =>
+{
+  React.useEffect(() => {
+    const elems = document.querySelectorAll('#sorting');
+    const instances = M.FormSelect.init(elems, {})
+  }, []);
+
+  const [openModal, closeModal] = useModal()
+  
+  return (
+    <div>
+    { props.permissions.canAddRoom ?
+    <div className="col s12">
+      <ul style={{height: "50px", lineHeight: "45px", margin: 0}}>
+        <li style={{float:"left", margin: "5px"}}>
+          <a href="#" className="waves-effect waves-light btn" onClick={
+            () => openModal(AddRoomModal, {closeModal, submit: create_room})
+          }> Add room </a>
+        </li>
+      </ul>
+    </div> : "" }
+    <div className="col s12">
+      <ul style={{height: "50px", lineHeight: "45px", "marginTop": 0}}>
+        <li style={{float: "left", margin: "5px", marginTop: "12px"}}>
+            <i className="material-icons black-text">search</i>
+        </li>
+        <li style={{float: "left", margin: "5px"}}>
+          <input type="text" placeholder="search" onChange={props.onSearch}/>
+        </li>
+        <li className="hide-on-med-and-down"style={{float: "left", margin: "5px"}}>
+          <label>
+            <input type="checkbox" onChange={props.onShowFullRoomsToggle} defaultChecked={true}/>
```

Comment:
Why all rooms disappear when toggling this off?

Reply:
I fixed it in https://github.com/ksiuwr/zosia16-site/pull/48/commits/b8607c8ec32374565e8ef89d7db9a98ee08d6694 - funny bug -> this code was unreachable and this function return nothing instead of true/false value